### ISSUE TITLE
Some messages appear out of order

### ISSUE THREAD

**Issue by vthriller** [vector-im/riot-web#6780]
### Description

Some messages appear out of order, like on this screenshot from `#mozilla_#servo:matrix.org`:

![image](https://user-images.githubusercontent.com/176062/40373439-8f60aee8-5def-11e8-84f4-e7791cf7d97f.png)

The following are the snippets of said messages sources:
```json
{
  "origin_server_ts": 1526498974801,
  "sender": "@mozilla_crowbot:matrix.org",
  …
  "age": 503879905,
  "unsigned": {
    "age": 503879905
  },
  …
}
```

```json
{
  "origin_server_ts": 1526498969477,
  "sender": "@mozilla_jdm:matrix.org",
  …
  "age": 503885229,
  "unsigned": {
    "age": 503885229
  },
  …
}
```

And, obviously,

```python
>>> 1526498974801 > 1526498969477
True
>>> 503879905 < 503885229
True
```

### Steps to reproduce

- Join something.
- Accumulate a week or so of unread conversations.
- Read all the messages.
- Try again if failed to reproduce.

I'm able to replicate this by just looking for `20472`, or `20632`, or `uluru/pull/8` in  `#mozilla_#servo:matrix.org`'s history (which, unfortunately, is not available for those who just joined the room).

**Log**: not sent

### Version information

- **Platform**: web
- **Browser**: Firefox 57
- **OS**: Gentoo Linux
- **URL**: riot.im/app (0.15.2 at the time of writing this issue)

**Comment by lampholder** [comment]
_As I understand it_ this is, as @t3chguy says, an unavoidable tradeoff and so I'm afraid not something that is going to be changed.

**Comment by cjdp** [comment]
Good day,

We are seeing some messages appear out of order between the riot iOS app and the riot web client.

We are hosting our own synapse home server (0.99.5.1+stretch1) on Debian 9. We also host our own riot web client (v1.1.2).

The browser used to access the web client is firefox-esr (60.7.0esr-1~deb9u1), also running on Debian 9.

Our home server is configured not to federate (both in configuration and firewall settings). We only have a single home server, so there is no chance of two home servers disagreeing.

This seems to be a bug.

**Comment by turt2live** [comment]
1.1.2 has a bug which should be fixed in 1.2.0 (due to be released soon - there is a RC available).

I'm actually going to close this issue as there's nothing actionable on it.
